extensions/s3/src/main/java/io/deephaven/extensions/s3/S3Instructions.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.extensions.s3;
+
+import io.deephaven.annotations.BuildableStyle;
+import org.immutables.value.Value;
+
+import java.time.Duration;
+
+/**
+ * This class provides instructions intended for reading and writing data to AWS S3 instances.
+ */
+@Value.Immutable
+@BuildableStyle
+public abstract class S3Instructions {
+
+    private final static int DEFAULT_MAX_CONCURRENT_REQUESTS = 50;
+    private final static int DEFAULT_READ_AHEAD_COUNT = 1;
+    private final static int DEFAULT_FRAGMENT_SIZE = 512 << 20; // 5 MB
```

Comment:
https://docs.aws.amazon.com/whitepapers/latest/s3-optimizing-performance-best-practices/use-byte-range-fetches.html
Suggests 8MB or 16MB is best practice. Does it materially alter performance?

Reply:
I can test this in my final benchmarks to set the default values.